docs/rfc/rfc-1.adoc

```diff
@@ -287,3 +287,42 @@ will begin to be slashed and the challenger will be rewarded.
 ===
 
 stub!
+
+[[bitcoin-sweeping-fee]]
+=== Bitcoin Sweeping Fee
+
+Any time a bitcoin transaction needs to be posted and then mined on the bitcoin
+blockchain, the miners need to be paid a fee for their work. This fee
+fluctuates with market demand and is decently volatile.
+
+When the depositor posts their address reveal on ethereum, they include a
+bitcoin `max_fee` denominated in total BTC that they're willing to pay for
+three different fees:
+
+- the over-the-bridge percentage cut (say 0.2% or whatever governance decides)
+- the over-the-bridge fixed cut (say 0.001 mBTC or whatever governance decides)
+  to deter folks from depositing dust
+- the mining fee for getting the batch sweeping transaction to post
+
+Next, we hash the most recent ethereum block to generate the seed for a random
+number generator to generate a modulus that selects a random unlucky operator
+that needs to check the blockstream api for the 25-block target fee:
+https://blockstream.info/api/fee-estimates and then post that information to
+the ethereum chain. The 25-block target fee from blockstream is per vByte, so
+we need to know how many vBytes will be in the transaction, which will depend
+on how many inputs will be in the transaction, which will depend on how many
+deposits we will sweep, which will depend on how many deposits find the target
+fee palatable.
```

Comment:
Yeah - we should have the unlucky operator publish the target fee on-chain

Reply:
to elaborate a bit - and maybe I'm missing what your comment is getting at:

we need to have the unlucky operator post the target fee on chain, otherwise we're re-inventing some form of a consensus algorithm when we already have a working one (the ethereum chain).

once we have consensus on what the fee is, everyone's client performs the same bin-packing algorithm to determine which inputs can be packed into the transaction, then they all sign that transaction and anyone broadcasts it

Reply:
I am not sure if the issue here is about consensus or about signature fraud. Let's say that we go with this problem as currently proposed. The question is: can we trust that the fee published on-chain is correct? What if they publish something that is way too low and it will take weeks to sweep the TX, incrementing 1.1x, 1.2x, ...

I think we already have some sort of consensus required. That is, all clients need to perform the same algorithm to select UTXOs for sweeping. In this case, why don't we have all clients checking the fee with blockstream? 
Alternatively, if we want to have one client proposing the batch, why don't we have this client proposing the fee as well? The validation could be as complex as running the UTXO picking algorithm by everyone to confirm the proposition is valid or, just validating that the proposed fee is +-10% from what the blockstream currently proposes and validating that the number of UTXOs included is +-10% of what is expected and that all UTXOs are valid for this sweeping session.

This part scares me less than signature frauds - we need to know that what the wallet signed was what that wallet was supposed to sign and the verification should be doable on-chain. This requires knowing what is in the batch and what was the fee along the way. 

We can also use an alternative approach. Instead of checking against particular bytes to be signed, we can check where the funds were moved just like you suggested some time ago but I am not sure how feasible on-chain this is.